src/blog/on-removing-let-let-mut.vmd

```diff
@@ -0,0 +1,330 @@
+---
+title: On Removing Let and Let Mut
+author: Evan Ovadia
+date: Apr 17, 2022
+realm: blog
+path: blog/on-removing-let-let-mut
+layout: annotated
+namespace: c-blog m-annotated
+sponsor: us
+---
+
+We're going to commit a cardinal sin today and talk about *syntax design!* [# Normally we talk about new memory management schemes, yak shaving contests, and other nonsense. It feels weird to be talking about something other people usually talk about!]
+
+
+<div><img src="/images/bikeshed.jpg" width="100%" style="opacity: .8"/><center><span style="opacity: .8"><a href="https://en.wikipedia.org/wiki/Law_of_triviality">(Also, while we have you here, what color should we paint this shed?)</a></span></center></div>
+
+
+We often think that there are only three alternatives for variable declaration:
+
+ * One keyword to declare a variable, such as:
+    * In C++, `auto x = 4;`
+    * Swift's `var x = 4`
+    * Go's `x := 4` (though it's not really a keyword per se)
+ * Specifying whether the variable can change, such as:
+    * Rust's `let x = 4;` vs `let mut x = 4;`
+    * Javascript's `let x = 4` vs `const x = 4`
+ * Leaving it off completely, such as:
+    * Python's `x = 4`
+
+
+The previous version of [Vale](https://vale.dev/) used the second option, `let` and `let mut`. [# Vale is a language that aims to be fast and memory-safe, while still being easy to learn. This syntax change helps with that!]
+
+However, it recently changed to a fourth option, which was so nice that it became the default behavior in version 0.2.
+
+
+<slice/>
+
+
+# The Fourth Option
+
+Most languages can't just use `x = 4;`, because that already meant something; that's the *re-assignment statement*, which modifies an existing variable that we already declared.
+
+And alas, Python tried combining those two statements. It didn't go well. [# This leads to problems; if you rename your `x = 1;` declaration to `y = 1;`, but forget to modify `x = 42;` re-assignment below, you now accidentally have two variables!]
+
+
+However, there's another option here: *let's change the re-assignment statement!*
+
+
+<<<<
+Instead of this:
+
+```vale
+func main() {
+  let a = 3;
+  let b = 3;
+  let c = 3;
+  let mut d = 3;
+  d = 7;
+  println(d);
+}
+```stdout
+7
+```
+////
+We can have this:
+
+```vale
+func main() {
+  a = 3;
+  b = 3;
+  c = 3;
+  d = 3;
+  set d = 7;
+  println(d);
+}
+```stdout
+7
+```
+>>>>
+
+In other words, `x = 3;` declares a variable, and `set x = 42;` re-assigns it.
+
+
+It was odd at first, but after using it for a few weeks, we think this is a huge improvement.
+
+
+Syntax design can be a tricky endeavor. When exploring new syntax, we had to suppress the knee-jerk "unfamiliarity avoidance", and actually experiment. Without experimenting, it's easy to get stuck with what's familiar, even if there are better options.
```

Comment:
Why is "unfamiliarity avoidance" in quotes? It doesn't seem to be a known term to me even in Vale land